GoogleDataTransport/GDTCORLibrary/GDTCORFlatFileStorage.m

```diff
@@ -126,14 +141,43 @@ - (void)storeEvent:(GDTCOREvent *)event
                                                expirationDate:event.expirationDate
                                                     mappingID:event.mappingID];
     NSError *error;
-    GDTCOREncodeArchive(event, filePath, &error);
+    NSData *encodedEvent = GDTCOREncodeArchive(event, nil, &error);
+    if (error) {
+      completion(NO, error);
+      return;
+    }
+
+    // Check storage size limit before storing the event.
+    uint64_t resultingStorageSize = self.sizeCalculator.directoryContentSize + encodedEvent.length;
+    if (resultingStorageSize > kGDTCORFlatFileStorageSizeLimit) {
+      NSError *error = [NSError
+          errorWithDomain:GDTCORFlatFileStorageErrorDomain
+                     code:GDTCORFlatFileStorageErrorSizeLimitReached
+                 userInfo:@{
+                   NSLocalizedFailureReasonErrorKey : @"Storage size limit has been reached."
+                 }];
+      completion(NO, error);
+      return;
+    }
+
+    // Write the encoded event to the file.
+    BOOL writeResult = GDTCORWriteDataToFile(encodedEvent, filePath, &error);
+    if (writeResult == NO || error) {
+      GDTCORLogDebug(@"Attempt to write archive failed: path:%@ error:%@", filePath, error);
+    } else {
+      GDTCORLogDebug(@"Writing archive succeeded: %@", filePath);
+    }
+
     if (error) {
       completion(NO, error);
       return;
     } else {
       completion(YES, nil);
     }
 
+    // Notify size calculator.
+    [self.sizeCalculator fileWithSize:encodedEvent.length wasAddedAtPath:filePath];
```

Comment:
Instead of a two step process of adding the event to the filepath and then updating the size. Can we update the size of the directory in the GDTCORWriteDataToFile() method instead?

Reply:
Currently `GDTCORWriteDataToFile()` is a generic and independent function that has no context on the file it stores, so updating the size tracker in it doesn't seem appropriate. 

But I like the idea in general. I've considered implementing something like `GDTFileManager` which will be responsible for file system operations and keeping track of directory sizes as needed. But this change will increase the scope of the PR, so I would prefer to defer it and implement later. WDYT?

Reply:
I agree, that goes outside the scope of this PR. How are we managing TODOs for Firelog? Consider creating an internal bug to track this.

Reply:
Created b/168098576